scripts/ligandtocore/QD.py

```diff
@@ -0,0 +1,223 @@
+        database_entry = False
+    else:
+        # Export the unoptimized ligand to a .pdb and .xyz file
+        ligand_name = 'ligand_' + ligand.get_formula()
+        molkit.writepdb(ligand, os.path.join(ligand_folder, ligand_name + '.pdb'))
+        ligand.write(os.path.join(ligand_folder, ligand_name + '.xyz'))
+        print('Ligand:\t\t\t\t' + str(ligand_name) + '.pdb')
+
+        if opt:
+            # Optimize the ligand
+            ligand = qd_scripts.global_minimum(ligand, ligand_folder)
+
+            # Set pdb_info
+            qd_scripts.set_pdb(ligand, 'LIG', is_core=False)
+
+            # Export the optimized ligand to a .pdb and .xyz file
+            molkit.writepdb(ligand, os.path.join(ligand_folder, ligand_name + '.opt.pdb'))
+            ligand.write(os.path.join(ligand_folder, ligand_name + '.opt.xyz'))
+            print('Optimized ligand:\t\t' + str(ligand_name) + '.opt.pdb')
+
+        # Create an entry for in the database
+        database_entry = qd_scripts.create_entry(ligand, opt)
+
+    # Identify functional groups within the ligand.
+    ligand_list, ligand_indices = qd_scripts.find_substructure(ligand, split)
+    for ligand in ligand_list:
+        ligand.add_atom(Atom(atnum=0, coords=ligand.get_center_of_mass()))
+
+    return ligand_list, ligand_indices, database_entry
+
+
+def prep_qd(core, ligand, core_indices, ligand_index, qd_folder):
+    """
+    Function that handles all quantum dot (qd, i.e. core + all ligands) operations.
+    """
+    # Rotate and translate all ligands to their position on the core.
+    # Returns a list of PLAMS molecules and atomic indices.
+    core = copy.deepcopy(core)
+    ligand_list = [qd_scripts.rotate_ligand(core, ligand, core_index, ligand_index, i)
+                   for i, core_index in enumerate(core_indices)]
+
+    ligand_list, ligand_indices = zip(*ligand_list)
+    core.delete_atom(core[-1])
+
+    # Prepare the .pdb filename as a string.
+    core_name = core.get_formula()
+    ligand_formula = ligand_list[0].get_formula()
+    ligand_heteroatom = ligand[ligand_index].symbol
+    ligand_name = ligand_formula + '_@_' + ligand_heteroatom + str(ligand_index)
+    pdb_name = str('core_' + core_name + '___ligand_' + ligand_name)
+
+    # Attach the rotated ligands to the core, returning the resulting strucutre (PLAMS Molecule).
+    qd = qd_scripts.combine_qd(core, ligand_list)
+
+    # indices of all the atoms in the core and the ligand heteroatom anchor.
+    qd_indices = [qd.atoms.index(atom) + 1 for atom in ligand_indices]
+    qd_indices += [i + 1 for i, atom in enumerate(core)]
+
+    qd.write(os.path.join(qd_folder, pdb_name + '.xyz'))
+    molkit.writepdb(qd, os.path.join(qd_folder, pdb_name + '.pdb'))
+    print('core + ligands:\t\t\t' + pdb_name + '.pdb')
+
+    return qd, pdb_name, qd_indices
+
+
+def prep_prep(path, dir_name_list, input_cores, input_ligands, smiles_extension, column, row,
```

Comment:
@BvB93 There are too many functions arguments, it would be great if you pass to the function a dictionary-liked structure containing all the data.